src/System.Text.Json/src/System/Text/Json/Serialization/Converters/DefaultArrayConverter.cs

```diff
@@ -11,7 +11,28 @@ internal sealed class DefaultArrayConverter : JsonEnumerableConverter
     {
         public override IEnumerable CreateFromList(Type enumerableType, Type elementType, IList sourceList)
         {
-            return CreateArrayFromList(elementType, sourceList);
+            Array array;
```

Comment:
Per other comments we shouldn't have to copy into an array.

Reply:
This is the `DefaultArrayConverter`. I used `IEnumerable<T>` elsewhere as discussed in other comments.